src/test/java/org/jenkinsci/plugins/gitclient/JGitClientTest.java

```diff
@@ -0,0 +1,162 @@
+package org.jenkinsci.plugins.gitclient;
+
+import hudson.EnvVars;
+import hudson.model.TaskListener;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import hudson.plugins.git.GitException;
+import hudson.plugins.git.GitObject;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+import java.util.Set;
+
+// Collides with implicit org.jenkinsci.plugins.gitclient.Git.
+//import org.eclipse.jgit.api.Git;
+import static org.eclipse.jgit.lib.Constants.HEAD;
+import org.eclipse.jgit.internal.storage.file.FileRepository;
+import org.eclipse.jgit.internal.storage.file.GC;
+import org.eclipse.jgit.lib.ObjectId;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import org.jvnet.hudson.test.Issue;
+
+/**
+ * JGit Client-specific tests. In their own Test file to ward off bloat in 
+ * GitClient and minimized parameterized and other test logic conditional 
+ * on Git implementation.
+ *
+ * @author Brian Ray
+ */
+public class JGitClientTest {
+    /* These tests are only for the JGit client. */
+    private static final String GIT_IMPL_NAME = "jgit";
+
+    /* Instance under test. */
+    private GitClient gitClient;
+
+    /* Lower level "porcelain" API when gitClient can't do something. */
+    private org.eclipse.jgit.api.Git gitAPI;
+
+    /* Repo garbage collector. */
+    private GC gc;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+    private File repoRoot;
+
+    @Before
+    public void setGitClientEtc() throws IOException, InterruptedException {
+        repoRoot = tempFolder.newFolder();
+        gitClient = Git.with(TaskListener.NULL, new EnvVars())
+                       .in(repoRoot)
+                       .using(GIT_IMPL_NAME)
+                       .getClient();
+        FileRepository repo = (FileRepository) gitClient.getRepository();
```

Comment:
I'd prefer that the test not use the deprecated `getRepository()` API.  It was deprecated long ago and in this case, it is easier to use the `repoRoot` File that is known to point to the root of the git repository.

If necessary, it could be moved after the `gitClient.init_()` call so that it is looking at a repository that is already initialized.
```suggestion
        FileRepositoryBuilder builder = new FileRepositoryBuilder();
        Repository repo = builder.setGitDir(repoRoot).findGitDir().build();
        // Repository repository = builder.setGitDir(new File(repoRoot, ".git")).findGitDir().build();
```